### DISCUSSION TITLE
How to proxy from Nginx already on my vps?

### DISCUSSION THREAD

**Discussion by sunjianan9900** [appwrite/appwrite#4329] [Q&A] [answered]
appwrite running on localhost:9080 and 9443
How can i add config to nginx ?

When I try to proxy the domain to port 9080, I can open it, but I can't display the static files, and none of the js files can be loaded

**Comment by IzenDev** [comment, marked as answer]
@sunjianan9900 

Add this in your `/etc/nginx/sites-available/default`.

```
server {

    server_name [YOUR_SERVER_NAME];

    location / {
        add_header          Strict-Transport-Security "max-age=31536000" always;
        add_header          X-Frame-Options deny;
        proxy_pass          https://localhost:9443;
        proxy_next_upstream error timeout invalid_header http_500 http_502 http_503 http_504;
        proxy_redirect      off;
        proxy_buffering     off;
        proxy_set_header    Host $host;
        proxy_set_header    X-Real-IP $remote_addr;
        proxy_set_header    X-Forwarded-For $proxy_add_x_forwarded_for;
    }
}
```

**Reply by drexhacker** [reply]
Works Well with HTTP but returns ERR_TOO_MANY_REDIRECTS in https

**Reply by stnguyen90** [reply]
@drexhacker, that's weird...everything is already https so I don't see what would be redirecting to https. 🧐

**Reply by stnguyen90** [reply]
In order to ensure realtime works, you'll also need to include the following directives:

```
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
```

**Comment by eldadfux** [comment]
For reference, there's also this post: https://dev.to/appwrite/learn-how-to-run-appwrite-with-your-own-custom-proxy-or-load-balancer-28k

**Comment by stnguyen90** [comment]
For those using Nginx Proxy Manager, I think [this](https://medium.com/@stnguyen90/how-to-run-appwrite-behind-nginx-19348ed34243) is the best setup.

**Reply by LeiYangGH** [reply]
how to reverse proxy appwrite only in a subpath? for example, i want https://localhost:443/appwrite
